### PR TITLE
Solace Read connector: SolaceIO PTransform init

### PR DESCRIPTION
**Pull request by bzablocki**
PR for the Solace Read connector (addresses https://github.com/apache/beam/issues/31440).
Initializes SolaceIO Ptransform. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -96,6 +96,8 @@
 
 ## I/Os
 
+* Support for X source added (Java/Python) ([#X](https://github.com/apache/beam/issues/X)).
+* Support for reading from Solace message broker added (Java) ([#31440](https://github.com/apache/beam/issues/31440)).
```
Comment (Abacn) [inline CHANGES.md:100]: We don't announce it before we start publish javadoc

Comment (bzablocki) [reply CHANGES.md:100]: ok!

**Comment by bzablocki** [comment]
Run Java PreCommit

File: sdks/java/io/solace/build.gradle [hunk 2]
```diff
@@ -30,4 +30,9 @@ ext.summary = """IO to read and write to Solace destinations (queues and topics)
 
 dependencies {
     implementation enforcedPlatform(library.java.google_cloud_platform_libraries_bom)
-}
\ No newline at end of file
+    implementation library.java.vendored_guava_32_1_2_jre
+    implementation project(path: ":sdks:java:core", configuration: "shadow")
+    implementation library.java.joda_time
+    implementation library.java.solace
+    implementation library.java.vendored_guava_32_1_2_jre
```
Comment (Abacn) [inline sdks/java/io/solace/build.gradle:37]:
```suggestion
```

File: sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/SolaceIO.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+     *   <li>create a {@link org.apache.beam.sdk.io.solace.broker.MessageReceiver}.
+     * </ul>
+     *
+     * <p>An existing implementation of the SempClientFactory includes {@link
+     * org.apache.beam.sdk.io.solace.broker.BasicAuthJcsmpSessionService} which implements the Basic
+     * Authentication to Solace. *
+     *
+     * <p>To use it, specify the credentials with the builder methods. *
+     *
+     * <p>The host is the IPv4 or IPv6 or host name of the appliance. IPv5 addresses must be encoded
```
Comment (Abacn) [inline sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/SolaceIO.java:160]:
```suggestion
     * <p>The host is the IPv4 or IPv6 or host name of the appliance. IPv6 addresses must be encoded
```

**Review by Abacn — APPROVED** [review]

**Pull request merged**